File: lab3 binary search/D.py
n, m = [int(i) for i in input().split()]
x, cnt = 0, 0
arr, max, min = [], [], []
a = [[int(j) for j in input().split()] for i in range(n)]

for i in range(n):
    x = 0
    for j in range(m):
        if arr[j][i] > arr[x][i]:
            x = j
    max.append(arr[x][i])

for i in range(n):
    x = 0
    for j in range(m):
        if arr[i][j] < arr[i][x]:
            x = j
    min.append(arr[i][x])

for i in range(n):
    for j in range(m):
        if arr[i][j] == min[i] and arr[i][j] == max[j]:
            cnt += 1

print(cnt)

File: lab3 binary search/D.cpp
#include <iostream>
using namespace std;
int main()
{
  int n, m;
    cin >> n >> m;
    int arr[n][m],mini[n],maxi[m];
  for(int i = 0; i < n; i++){
        mini[i] = 1e5;
    }
  for(int j = 0; j < m; j++){
        maxi[j] = -1e5;
    }
  for(int i = 0;i < n; i++){
        for(int j = 0; j < m; j++){
            cin >> arr[i][j];
            mini[i]=min(mini[i], arr[i][j]);
            maxi[j]=max(maxi[j], arr[i][j]);
        }
    }
    int cnt = 0;
    for(int i = 0; i < n; i++){
        for(int j = 0;j < m; j++){
            if(arr[i][j] == mini[i] && arr[i][j] == maxi[j]){
                cnt++;
            }
        }
    }
    cout<<cnt;
}